src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/MvcDataAnnotationsMvcOptionsSetup.cs

```diff
@@ -11,28 +11,63 @@ namespace Microsoft.AspNetCore.Mvc.DataAnnotations.Internal
     /// <summary>
     /// Sets up default options for <see cref="MvcOptions"/>.
     /// </summary>
-    public class MvcDataAnnotationsMvcOptionsSetup : ConfigureOptions<MvcOptions>
+    public class MvcDataAnnotationsMvcOptionsSetup : IConfigureOptions<MvcOptions>
     {
-        public MvcDataAnnotationsMvcOptionsSetup(IServiceProvider serviceProvider)
-            : base(options => ConfigureMvc(options, serviceProvider))
+        private IStringLocalizerFactory _stringLocalizerFactory;
+        private IValidationAttributeAdapterProvider _validationAttributeAdapterProvider;
+        private IOptions<MvcDataAnnotationsLocalizationOptions> _dataAnnotationLocalizationOptions;
+
+        public MvcDataAnnotationsMvcOptionsSetup(
+            IValidationAttributeAdapterProvider validationAttributeAdapterProvider,
+            IOptions<MvcDataAnnotationsLocalizationOptions> dataAnnotationLocalizationOptions)
         {
+            if (validationAttributeAdapterProvider == null)
+            {
+                throw new ArgumentNullException(nameof(validationAttributeAdapterProvider));
+            }
+
+            if (dataAnnotationLocalizationOptions == null)
+            {
+                throw new ArgumentNullException(nameof(dataAnnotationLocalizationOptions));
+            }
+
+            _validationAttributeAdapterProvider = validationAttributeAdapterProvider;
+            _dataAnnotationLocalizationOptions = dataAnnotationLocalizationOptions;
         }
 
-        public static void ConfigureMvc(MvcOptions options, IServiceProvider serviceProvider)
+        public MvcDataAnnotationsMvcOptionsSetup(
+            IValidationAttributeAdapterProvider validationAttributeAdapterProvider,
+            IOptions<MvcDataAnnotationsLocalizationOptions> dataAnnotationLocalizationOptions,
+            IStringLocalizerFactory stringLocalizerFactory)
```

Comment:
`: this(validationAttributeAdapterProvider, dataAnnotationLocalizationOptions)`. You could remove the null checking from here in that case.